radixdlt/babylon-wallet-android
# [ABW-2754] Show add link connector screen only when no link connectors available and connector status indicator

## Description
This PR 
- fixes when showing the add new link connector screen. Specifically, wallet shows that screen only when no link connectors configured in settings. We do not check if the linked connectors are connected any more.
- added a loading spinner in the "add ledger device" button while user is linking a new connector

### For dev and alpha builds only:
Added a peer connection status indicator (not data channel status!) which is shown as a dot at the top of the system's bar.
- indicator turns **red** as long as the peer connection is in **disconnected** state
- indicator turns **yellow** as long as the peer connection is in **connecting** state
- indicator turns **green** as long as the peer connection is in **connected** state


## How to test

1. Add a new link connector
2. Forget link connection from the CE of the browser, but do not delete it from the wallet settings
3. Navigate to Ledger Hardware Wallets screen and click on "Add Ledger Device"
=> it should show the "Add Ledger Device" screen

now continue with:
1. Delete the linked connector from settings
2. Navigate again back to Ledger Hardware Wallets screen and click on "Add Ledger Device"
=> it should show the add new link connector screen


## Video

https://github.com/radixdlt/babylon-wallet-android/assets/118305718/e044a0c4-726b-4384-a948-74df88e3eb16


## PR submission checklist
- [x] I have tested all the paths where ledger and linked connector are involved
